perl-lib/OESS/lib/OESS/MPLS/Device/Juniper/MX.pm

```diff
@@ -188,6 +199,19 @@ sub connected{
     return $self->{'connected'};
 }
 
+sub verify_connection{
+    #gather basic system information needed later, and make sure it is what we expected / are prepared to handle                                                                            
+    my $self = shift;
+    my $sysinfo = $self->get_system_information();
+    if (($self->{"os_name"} = "junos") && ($self->{"version_name"} = "13.3R1.6")){
```

Comment:
This is not how we compare string values in perl :)

you should use eq or ne when doing string compares